waku/waku_rendezvous/waku_peer_record.nim

```diff
@@ -0,0 +1,74 @@
+import std/times, sugar
+
+import
+  libp2p/[
+    protocols/rendezvous,
+    signed_envelope,
+    multicodec,
+    multiaddress,
+    protobuf/minprotobuf,
+    peerid,
+  ]
+
+type WakuPeerRecord* = object
+  # Considering only mix as of now, but we can keep extending this to include all capabilities part of Waku ENR
+  peerId*: PeerId
+  seqNo*: uint64
+  addresses*: seq[MultiAddress]
+  mixKey*: string
+
+proc payloadDomain*(T: typedesc[WakuPeerRecord]): string =
+  $multiCodec("libp2p-custom-peer-record")
+
+proc payloadType*(T: typedesc[WakuPeerRecord]): seq[byte] =
```

Comment:
With quick search I can't find this defined anywhere or in the standard multicodec table - is this our own payload type for "custom peer record" or specifically `WakuPeerRecord`? If so, can we make this a `const` or add it to the spec?